### PR TITLE
Replace State class with reusable Context class

### PR DESCRIPTION
**Pull request by apasel422**
In addition to benefiting the structured-header validators, this will make it easier to propagate data such as vendor-specific values to the JSON validators in subsequent changes.

### CODE REVIEW

File: header-validator/src/validate-os.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
-import { Issue, ValidationResult } from './issue'
-const { parseList } = require('structured-headers')
+import { Context, ValidationResult } from './context'
+import { InnerList, Item, parseList } from 'structured-headers'
 
-function validateURL(item: any): string|undefined {
-  if (typeof item !== 'string') {
-    return 'must be a string'
+function validateURL(ctx: Context, member: InnerList|Item): void {
+  if (typeof member[0] !== 'string') {
```
Comment (agarant) [inline header-validator/src/validate-os.ts:5]: Might be useful to spread the values in the member tuple to document what they represent.

Comment (agarant) [inline header-validator/src/validate-os.ts:5]: Might be useful to spread the values in the member tuple to document what they represent.

**Review by agarant — APPROVED** [review]
nice!

File: header-validator/src/validate-os.ts (hunk 1 above)
Comment (apasel422) [reply header-validator/src/validate-os.ts:5]: Ack. It's a bit difficult to do so because of the weird typing there, but will pursue in a followup.

**Pull request merged**